include/envoy/stats/timespan.h

```diff
@@ -18,43 +15,18 @@ class CompletableTimespan {
   virtual ~CompletableTimespan() = default;
 
   /**
-   * Complete the timespan.
-   */
-  virtual void complete() PURE;
-};
-
-/**
- * An individual timespan that flushes its measured value in time unit (e.g
- * std::chrono::milliseconds). The initial time is captured on construction. A timespan must be
- * completed via complete() for it to be stored. If the timespan is deleted this will be treated as
- * a cancellation.
- */
-template <class TimeUnit> class TimespanWithUnit : public CompletableTimespan {
-public:
-  TimespanWithUnit(Histogram& histogram, TimeSource& time_source)
-      : time_source_(time_source), histogram_(histogram), start_(time_source.monotonicTime()) {}
-
-  /**
-   * Complete the timespan and send the time to the histogram.
+   * Time elapsed since the creation of the timespan, in milliseconds.
    */
-  void complete() override { histogram_.recordValue(getRawDuration().count()); }
+  virtual uint64_t elapsedMs() PURE;
```

Comment:
Can this return a `std::chrono::millseconds` and just be called `elapsed()`?

Reply:
Good point, fixed.